core/src/trezor/ui/layouts/tt_v2/reset.py

```diff
@@ -13,43 +39,303 @@ async def show_share_words(
+                # change the counter either so just use the initial value.
+                result = (result, count)
+        status, value = result
+
+        if status == trezorui2.CONFIRMED:
+            assert isinstance(value, int)
+            return value
+
+        await ctx.wait(
+            _RustLayout(
+                trezorui2.show_simple(
+                    title=None, description=info(value), button="OK, I UNDERSTAND"
+                )
+            )
+        )
+        num_input.request_complete_repaint()
 
 
 async def slip39_prompt_threshold(
     ctx: wire.GenericContext, num_of_shares: int, group_id: int | None = None
 ) -> int:
-    raise NotImplementedError
+    count = num_of_shares // 2 + 1
+    # min value of share threshold is 2 unless the number of shares is 1
+    # number of shares 1 is possible in advanced slip39
+    min_count = min(2, num_of_shares)
+    max_count = num_of_shares
+
+    def description(count: int):
+        if group_id is None:
+            if count == 1:
+                return "For recovery you need 1 share."
+            elif count == max_count:
+                return f"For recovery you need all {count} of the shares."
+            else:
+                return f"For recovery you need any {count} of the shares."
+        else:
+            return f"The required number of shares to form Group {group_id + 1}."
+
+    def info(count: int):
+        text = "The threshold sets the number of shares "
+        if group_id is None:
+            text += "needed to recover your wallet. "
+            text += f"Set it to {count} and you will need "
+            if num_of_shares == 1:
+                text += "1 share."
+            elif num_of_shares == count:
+                text += f"all {count} of your {num_of_shares} shares."
+            else:
+                text += f"any {count} of your {num_of_shares} shares."
+        else:
+            text += "needed to form a group. "
+            text += f"Set it to {count} and you will "
+            if num_of_shares == 1:
+                text += "need 1 share "
+            elif num_of_shares == count:
+                text += f"need all {count} of {num_of_shares} shares "
+            else:
+                text += f"need any {count} of {num_of_shares} shares "
+            text += f"to form Group {group_id + 1}."
+        return text
+
+    return await _prompt_number(
+        ctx,
+        title="SET THRESHOLD",
+        description=description,
+        info=info,
+        count=count,
+        min_count=min_count,
+        max_count=max_count,
+        br_name="slip39_threshold",
+    )
 
 
 async def slip39_prompt_number_of_shares(
     ctx: wire.GenericContext, group_id: int | None = None
 ) -> int:
-    raise NotImplementedError
+    count = 5
+    min_count = 1
+    max_count = 16
+
+    def description(i: int):
+        if group_id is None:
+            if i == 1:
+                return "Only one share willi be created."
```

Comment:
typo: willi

Reply:
uhh, fixed in 783acd7f0bf350ab3e99fd809df67ca79d2c4f03